src/test/ui/layout/unsafe-cell-hides-niche.rs

```diff
@@ -18,12 +19,23 @@ struct NoNiche<T>(UnsafeCell<T>);
 
 fn main() {
     assert_eq!(size_of::<Option<Wrapper<u32>>>(),     8);
-    assert_eq!(size_of::<Option<Wrapper<N32>>>(),     4);
+    assert_eq!(size_of::<Option<Wrapper<N32>>>(),     4); // (✓ niche opt)
     assert_eq!(size_of::<Option<Transparent<u32>>>(), 8);
-    assert_eq!(size_of::<Option<Transparent<N32>>>(), 4);
+    assert_eq!(size_of::<Option<Transparent<N32>>>(), 4); // (✓ niche opt)
     assert_eq!(size_of::<Option<NoNiche<u32>>>(),     8);
-    assert_eq!(size_of::<Option<NoNiche<N32>>>(),     8);
+    assert_eq!(size_of::<Option<NoNiche<N32>>>(),     8); // (✗ niche opt)
 
     assert_eq!(size_of::<Option<UnsafeCell<u32>>>(),  8);
-    assert_eq!(size_of::<Option<UnsafeCell<N32>>>(),  8);
+    assert_eq!(size_of::<Option<UnsafeCell<N32>>>(),  8); // (✗ niche opt)
+
+    assert_eq!(size_of::<       UnsafeCell<&()> >(), 8);
```

Comment:
How does this pass on 32bit targets... :thinking:

Reply:
Heh, yeah, the values should be replaced with e.g. `size_of::<usize>() * 3` instead of `24`.

(Could use a `const` for `size_of::<usize>()` instead of leaving it inline, tho I can't think of a good name)

Reply:
The thing is, CI passes on this PR, and that should run 32bit tests as well... right? Or does it just check-pass them all?

Reply:
> Or does it just check-pass them all?

Looks like it: https://github.com/rust-lang/rust/blob/06754d8852bea286a3a76d373ccd17e66afb5a8b/src/ci/docker/host-x86_64/x86_64-gnu-llvm-12/Dockerfile#L37-L51

(theoretically we could move asserts like these from `fn main() {...}` to a `const _: () = {...};` though only by redefining `assert_eq!($a:expr, $b:expr)` to be `assert!($a == $b)`, without the value printing)